lib/condor.py

```diff
@@ -227,12 +221,93 @@ def submit_dag(
                 sys.stderr.write(
                     f"Error: Child was terminated by signal {-output.returncode}"
                 )
-            else:
-                if "outdir" in vargs:
-                    print(
-                        f"Output will be in {vargs['outdir']} after running jobsub_transfer_data."
-                    )
         except OSError as e:
             print("Execution failed: ", e)
 
     return submit(subfile, vargs, schedd_name)
+
+
+class JobIdError(Exception):
+    pass
+
+
+class Job:
+    """
+    Job represents a single HTCondor batch job or cluster with an id like
+        <cluster>[.<process>]@<schedd>
+
+    where if <process> is specified it is a single job, or if not then a cluster
+    of jobs. Examples:
+
+
+        123@schedd.example.com
+        123.0@schedd.example.com
+        123.456@schedd.example.com
+
+    """
+
+    id: str
+    seq: int
+    proc: int
+    schedd: str
+    cluster: bool
+
+    _id_regexp = re.compile(r"(\d+)(?:\.(\d+))?@([\w\.]+)")
+
+    def __init__(self, job_id: str):
+        self.id = job_id
+        m = Job._id_regexp.match(job_id)
+        if m is None:
+            raise JobIdError(f'unable to parse job id "{job_id}"')
+        try:
+            self.seq = int(m.group(1))  # seq is required
+            if m.group(2) is None:  # proc is optional
+                self.cluster = True
+                self.proc = 0
+            else:
+                self.cluster = False
+                self.proc = int(m.group(2))
+            self.schedd = m.group(3)  # schedd is required
+        except TypeError as e:
+            raise JobIdError(f'error when parsing job id "{job_id}"') from e
```

Comment:
Nice - we need some more of these custom errors in the code as we move closer to production.

Reply:
Agreed. We should also discuss overall error handling and what to show the user - python tracebacks are informative to developers (and so good in tickets) but are not so helpful to the users. But, later...